components/script/dom/element.rs

```diff
@@ -1930,6 +1931,56 @@ impl Element {
         self.set_attribute(local_name, AttrValue::String(value.to_string()), can_gc);
     }
 
+    pub(crate) fn get_trusted_type_url_attribute(
+        &self,
+        local_name: &LocalName,
+    ) -> TrustedScriptURLOrUSVString {
+        assert!(*local_name == local_name.to_ascii_lowercase());
+        let attr = match self.get_attribute(&ns!(), local_name) {
+            Some(attr) => attr,
+            None => return TrustedScriptURLOrUSVString::USVString(USVString::default()),
+        };
+        let value = &**attr.value();
+        // XXXManishearth this doesn't handle `javascript:` urls properly
+        self.owner_document()
+            .base_url()
+            .join(value)
+            .map(|parsed| TrustedScriptURLOrUSVString::USVString(USVString(parsed.into_string())))
+            .unwrap_or_else(|_| TrustedScriptURLOrUSVString::USVString(USVString(value.to_owned())))
+    }
+
+    pub(crate) fn set_trusted_type_url_attribute(
+        &self,
+        local_name: &LocalName,
+        value: TrustedScriptURLOrUSVString,
+        can_gc: CanGc,
+    ) -> Fallible<()> {
+        assert!(*local_name == local_name.to_ascii_lowercase());
+        let value = match value {
+            TrustedScriptURLOrUSVString::USVString(url) => {
+                let global = self.owner_global();
+                // TODO(36258): Reflectively get the name of the class
+                let sink = format!("{} {}", "HTMLScriptElement", &local_name);
```

Comment:
How can I retrieve the JS class name of the element? I looked into `std::any::type_name`, but that returns the Rust type (and is not stable). The information is in WebIDL, as well as the name of the struct. But I don't know if we can pass that into the macro as a parameter or dynamically insert it?

Reply:
We do this in the console serialization code, and it's ugly: https://github.com/servo/servo/blob/d4824a186a1f208a0c8f68f90a42d2aa1815b215/components/script/dom/console.rs#L279

Reply:
Do you mind if we fix this in a follow-up PR? That sounds a bit more hairy than I would like to fix right now.